engine/src/main/java/org/terasology/rendering/dag/nodes/BufferClearingNode.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag.nodes;
+
+import org.terasology.rendering.dag.AbstractNode;
+import org.terasology.rendering.dag.stateChanges.BindFBO;
+import org.terasology.rendering.opengl.FBOConfig;
+
+import static org.lwjgl.opengl.GL11.glClear;
+
+/**
+ * Instances of this node clear specific buffers attached to an FBOs, in accordance to a clearing mask.
+ * Normally this means that all the pixels in the buffers selected by the mask are reset to a default value.
+ *
+ * Notice that the node is fully initialised and ready to use only after calling the initialise(Object object) method.
+ *
+ * This class could be inherited by a more specific class that sets the default values, via (yet to be written)
+ * state changes.
+ */
+public class BufferClearingNode extends AbstractNode {
+
+    private int clearingMask;
+
+    /**
+     * Throws a RuntimeException if invoked. Use initialise(Object data) instead.
+     */
+    public void initialise() {
+        throw new RuntimeException("Please do not use initialise(). For this class use initialise(Object initialData) instead.");
```

Comment:
Certainly looks effective! "Trapping" methods like this is valid when they shouldn't be used IIRC, although there is a balance vs. architecture that attempts to avoid invalid methods entirely instead.